bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/ModuleHandlerCallback.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.automation;
+
+import java.util.Map;
+
+import org.eclipse.smarthome.core.common.registry.RegistryChangeListener;
+
+/**
+ * This class is responsible to provide a {@link RegistryChangeListener} logic. A instance of it is added to
+ * {@link RuleRegistry} service, to listen for changes when a single {@link RuleImpl} has been added, updated, enabled,
+ * disabled or removed and to involve RuleImpl Engine to process these changes. Also to send a {@code run} command
+ * for a single {@link RuleImpl} to the RuleImpl Engine.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public interface ModuleHandlerCallback {
+
+    /**
+     * This method gets <b>enabled</b> {@link RuleStatus} for a {@link RuleImpl}.
+     * The <b>enabled</b> rule statuses are {@link RuleStatus#UNINITIALIZED}, {@link RuleStatus#IDLE} and
+     * {@link RuleStatus#RUNNING}.
+     * The <b>disabled</b> rule status is {@link RuleStatus#DISABLED}.
+     *
+     * @param ruleUID UID of the {@link RuleImpl}
+     * @return {@code true} when the {@link RuleStatus} is one of the {@link RuleStatus#UNINITIALIZED},
+     *         {@link RuleStatus#IDLE} and {@link RuleStatus#RUNNING}, {@code false} when it is
+     *         {@link RuleStatus#DISABLED} and {@code null} when it is not available.
+     */
+    public Boolean isEnabled(String ruleUID);
```

Comment:
If my reading of the implementation has been correct this method returns null if the rule itself cannot be found in the registry.
Are there other reasons that return null?
"when it is not available" -- what is referred by "it"? The rule in the registry? The status of a known rule (I assume every existing rule must have a status assigned)?
Another option would to raise a IAE if the rule does not exist.